flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/TableDescriptor.java

```diff
@@ -16,9 +16,14 @@
  * limitations under the License.
  */
 
-package org.apache.flink.table.descriptors
+package org.apache.flink.table.descriptors;
+
+import org.apache.flink.annotation.PublicEvolving;
 
 /**
   * Common class for all descriptors describing table sources and sinks.
   */
-abstract class TableDescriptor extends DescriptorBase
+@PublicEvolving
+public abstract class TableDescriptor extends DescriptorBase {
```

Comment:
Just a thought. How about we convert it to interface? That means converting `DescriptorBase` to interface as well. What do you think @twalthr about it?

Reply:
Hi, @dawidwys , thanks for review. I updated the code except this place right now. 
It looks `DescriptorBase` overrides `toString()` method implementation, if it is converted to interface how to migrate this method...